fabfile.py

```diff
@@ -370,35 +370,69 @@ def reset_local_media(service_info_directory):
 
 
 @task
-def refresh_environment():
-    # NOTE: This command does not function yet.
-    # See: https://github.com/theirc/ServiceInfo-ircdeploy/pull/16#issue-125613744
+def refresh_environment(project_path=None):
+    """
+    Refresh any non-production environment with new DB data and new media.
+
+    If ``project_path`` is not provided, then we get a DB dump and media from production.
+
+    If ``project_path`` is provided, then we get the DB dump and media from inside the
+    ``project_path``. We expect the dump to be at ``${project_path}/service_info.sql`` and
+    the media to be at ``${project_path}/public/media``.
+    """
     require('environment')
     if env.environment == 'production':
         abort('Production cannot be refreshed!')
-    env.source_environment = 'production'
+    dump_file_name = '%(project)s.sql' % env
+    media_path = 'media'
+    db_name = db_user = '%s_%s' % (env.project, env.environment)
 
-    prod_db_name = '%(project)s_%(source_environment)s' % env
-    dump_file_name = '%s.sql' % prod_db_name
-    full_dump_file_path = os.path.join(env.project_root, dump_file_name)
-    prod_host = env.production_master
+    if project_path is None:
+        # We are refreshing from the live production server
+        prod_dump_file_path = os.path.join('/tmp', dump_file_name)
 
-    with settings(host_string=prod_host):
-        sudo('%s pg_dump -Ox %s -U %s > %s' % (env.db_wrapper, prod_db_name, prod_db_name, full_dump_file_path))
+        with settings(host_string=env.production_master):
+            prod_db_name = '%(project)s_production' % env
+            sudo('%s pg_dump -Ox %s -U %s > %s' % (env.db_wrapper, prod_db_name, prod_db_name, prod_dump_file_path))
+            get(prod_dump_file_path, dump_file_name)
+            get(env.media_source, media_path)
+    else:
+        # We are refreshing from files in the local project_path directory
+        current_dump_path = os.path.join(project_path, dump_file_name)
+        current_media_path = os.path.join(project_path, 'public', media_path)
+        local('cp %s %s' % (current_dump_path, dump_file_name))
+        local('cp -r %s .' % current_media_path)
 
-    db_name = db_user = '%s_%s' % (env.project, env.environment)
     with cd('/tmp'):
-        run('scp %s:%s %s' % (prod_host, full_dump_file_path, dump_file_name))
+        put(local_path=dump_file_name, remote_path=dump_file_name)
```

Comment:
Need to remove the file from /tmp when done. It's owned by the individual user, so another user won't be able to overwrite it if they try later to do another refresh.

Reply:
same for /tmp/media